provider/gce/environ_broker.go

```diff
@@ -308,6 +323,30 @@ func getDisks(imageURL string, cons constraints.Value, os ostype.OSType) ([]*com
 	return []*computepb.AttachedDisk{disk}, nil
 }
 
+// hasAccelerator checks if the given instance type has any accelerators (e.g., GPUs).
+func (env *environ) hasAccelerator(ctx context.ProviderCallContext, zone string, instanceType string) (bool, error) {
+	if len(instanceType) == 0 {
+		return false, nil
+	}
+
+	mt, err := env.gce.MachineType(ctx, zone, instanceType)
+	if err != nil {
+		return false, google.HandleCredentialError(errors.Trace(err), ctx)
+	}
+
+	accelerators := mt.GetAccelerators()
+	if len(accelerators) == 0 {
+		return false, nil
+	}
```

Comment:
Don't need this.
For loop can just be
`for _, accelerator := range mt.GetAccelerators() {`